### ISSUE TITLE
ask some questions about the paper

### ISSUE THREAD

**Issue by whzruc** [ggustavo/HotDataClassification#1]
Hello,

I hope to connect with you because I have come across your project, and I've also been studying a particular research paper recently. I would like to inquire if you have an understanding of the derivation of Equation 3 and if you could kindly explain it to me in detail. Your assistance would be greatly appreciated!

Best regards,
from haozhewang

**Comment by ggustavo** [comment]
Hello Haozhe Wang, it's a pleasure to try to assist you. I created this code a few years ago, and I don't remember it well. Also, I didn't get a chance to test if the implementation is correct.

I don't remember the details of how this equation works. Well, recently, I wrote a paper using Exponential Smoothing, but instead of using frequency, I used the distance between requests. I'll attempt to explain my idea; maybe this can help you understand a bit of [Levandoski](https://ieeexplore.ieee.org/author/37394404400)'s paper.

![ES](https://github.com/ggustavo/hot-data-classification/assets/6693953/64524622-ce6a-49f5-97d9-f679e5695d7f)

Imagine a set of page requests. Each page is capable of generating a time series and provides an estimate of when that page will be accessed in the near future. By selecting only page ID 4, which appears at times 1, 4, 5, 8, and 100, we can calculate the distances between subsequent requests for page 4. Let's forget the absolute error for now.

The basic formula is: 
![SE](https://github.com/ggustavo/hot-data-classification/assets/6693953/0a011c9c-f1a1-46b6-8284-8dd31dd81a03)
Where ES(t) is the expected prediction value, D(t) is the observed distance in the time series at time t, ES(t-1) is the previously calculated prediction value, and α (smoothing factor, ranging from 0 to 1) is an adjustment parameter for smoothing calibration.


Consider α = 0.5. 

- At time 1, it is not possible to calculate ES(1) because we do not have another request to calculate the distance for page 4.
- At time 4, since no previous prediction (ES(t-1)) has been calculated, we will assume, for this context, that the prediction value will be equal to the distance: ES(4) = 3.
- At time 5, we have the distance D(5) = 1, and the previous prediction ES(4) is assumed to be 3. Therefore, we can apply the model ES(5) = 3 + α (1 - 3) = 2.
- At time 8, we have the distance D(8) = 3, and the previous prediction ES(5) is 2. Thus, applying the model ES(8) = 2 + α (3 - 2) = 2.5.
- At time 100, we have the distance D(100) = 92, and the previous prediction ES(8) is 2.5. So, ES(100) = 2.5 + α (92 - 2.5) = 47.25.

In the end, the algorithm predicts that page 4 may be requested in the future after approximately 47.25 requests have occurred.
